pkg/sdk/grants_test.go

```diff
@@ -847,30 +847,40 @@ func TestRevokePrivilegeFromShare(t *testing.T) {
 	t.Run("on view", func(t *testing.T) {
 		otherID := RandomSchemaObjectIdentifier()
 		opts := &revokePrivilegeFromShareOptions{
-			privilege: ObjectPrivilegeUsage,
-			On: &RevokePrivilegeFromShareOn{
-				View: &OnView{
-					Name: otherID,
-				},
+			privileges: []ObjectPrivilege{ObjectPrivilegeUsage},
+			On: &ShareGrantOn{
+				View: otherID,
 			},
 			from: id,
 		}
 		assertOptsValidAndSQLEquals(t, opts, "REVOKE USAGE ON VIEW %s FROM SHARE %s", otherID.FullyQualifiedName(), id.FullyQualifiedName())
 	})
 
-	t.Run("on all views", func(t *testing.T) {
-		otherID := RandomDatabaseObjectIdentifier()
+	t.Run("on tag", func(t *testing.T) {
 		opts := &revokePrivilegeFromShareOptions{
-			privilege: ObjectPrivilegeUsage,
-			On: &RevokePrivilegeFromShareOn{
-				View: &OnView{
-					AllInSchema: otherID,
-				},
+			privileges: []ObjectPrivilege{ObjectPrivilegeRead},
+			On: &ShareGrantOn{
+				Tag: NewSchemaObjectIdentifier("database-name", "schema-name", "tag-name"),
 			},
 			from: id,
 		}
-		assertOptsValidAndSQLEquals(t, opts, "REVOKE USAGE ON ALL VIEWS IN SCHEMA %s FROM SHARE %s", otherID.FullyQualifiedName(), id.FullyQualifiedName())
-	})
+		assertOptsValidAndSQLEquals(t, opts, "REVOKE READ ON TAG  \"database-name\".\"schema-name\".\"tag-name\" FROM SHARE %s", id.FullyQualifiedName())
+	})
+
+	// TODO: This one throws an error
```

Comment:
what error? It throws it on unit or integration level? When we will address it?

Reply:
Also my mistake. This option was possible in the SDK but Snowflake is not allowing it (I mean `grant/revoke on all views`). Removed the test case.